File: javaArchive/src/IDs/StudentID.java
package IDs;

//Samet CAN 150120528
public class StudentID implements UniqueID {

	private int departmentCode;
	private int yearCode;
	private int orderOfPlacement;

	public StudentID(int departmentCode, int yearCode, int orderOfPlacement) {
		super();
		this.departmentCode = departmentCode;
		this.yearCode = yearCode;
		this.orderOfPlacement = orderOfPlacement;
	}
	
	public StudentID(String ID) {
		this.setID(ID);
	}

	// Set and get methods for ID
	public void setID(int DepartmentCode, int YearCode, int OrderOfPlacement) {
		this.departmentCode = DepartmentCode;
		this.yearCode = YearCode;
		this.orderOfPlacement = OrderOfPlacement;
	}

	public String getID() {
		return digitFixer(departmentCode) + digitFixer(yearCode) + digitFixer(orderOfPlacement);
	}

	// Another set method for ID
	public void setID(String string) {
		try {
			Integer.parseInt(string);
		} catch (NumberFormatException e) {

		} finally {
			this.departmentCode = Integer.parseInt(string.substring(0, 3));
			this.yearCode = Integer.parseInt(string.substring(3, 6));
			this.orderOfPlacement = Integer.parseInt(string.substring(6));
		}

	}

	// A method for giving fixed amount of numbers according to input number
	private String digitFixer(int integer) {
		String tempOrder = "" + integer;
		if (integer < 10) {
			tempOrder = "00" + tempOrder;
		} else if (integer < 100) {
			tempOrder = "0" + tempOrder;
		}
		return tempOrder;
	}
	

}

File: CSE3063F22P1_GRP1_iteration3/pythonCode/StudentID.py
from UniqueID import UniqueID

class StudentID(UniqueID):

    def __init__(self, *args):
        self.__departmentCode = 0
        self.__yearCode = 0
        self.__orderOfPlacement = 0
        index = 0
        for t in args: 
            if (isinstance(t, int)):
                if index == 0:
                    self.__departmentCode = t
                if index == 1:
                    self.__yearCode = t
                if index == 2:
                    self.__orderOfPlacement = t
            index += 1
            if (isinstance(t, str)):
                self.setID(t)

    # Creating properties for variables
        
    def setID(self, *args):
        index = 0
        for t in args: 
            if (isinstance(t, int)):
                if index == 0:
                    self.__departmentCode = t
                if index == 1:
                    self.__yearCode = t
                if index == 2:
                    self.__orderOfPlacement = t
            index += 1

            if (isinstance(t, str)):
                try:
                    int(t)
                except ValueError:
                    pass
                self.__departmentCode = int(t[:3])
                self.__yearCode = int(t[3:6])
                self.__orderOfPlacement = int(t[6:])

    def getID(self) -> str:
        return self.digitFixer(self.__departmentCode) + self.digitFixer(self.__yearCode) + self.digitFixer(self.__orderOfPlacement)

    ## Creating another methods
    def digitFixer(self, integer: int) -> str:
        tempOrder = str(integer)
        if integer < 10:
            tempOrder = "00" + tempOrder
        elif integer < 100:
            tempOrder = "0" + tempOrder
        return tempOrder
